lib/gooddata/helpers/auth_helpers.rb

```diff
@@ -31,13 +51,13 @@ def read_credentials(credentials_file_path = credentials_file)
         # @return [String] auth token from .gooddata, environment variable or nil
         def read_token(credentials_file_path = credentials_file)
           goodfile = read_credentials(credentials_file_path)
-          goodfile[:auth_token] || goodfile[:token] || ENV['GD_PROJECT_TOKEN']
+          [goodfile[:auth_token],  goodfile[:token], ENV['GD_PROJECT_TOKEN']].find { |x| !x.nil? && !x.empty? }
```

Comment:
One think I did not notice before. Do not we want precedence of ENV over .gooddata?